dataset/models/tf/layers/conv.py

```diff
@@ -96,8 +134,71 @@ def conv_transpose(inputs, filters, kernel_size, strides, *args, **kwargs):
     return output
 
 
+def _separable_conv(transpose, inputs, filters, kernel_size, strides, padding, data_format,
+                    dilation_rate, depth_multiplier, activation, name, **kwargs):
+    dim = inputs.shape.ndims - 2
+    context = None
+    if name is not None:
+        context = tf.variable_scope(name)
+        context.__enter__()
+    if transpose:
+        conv_layer = conv_transpose
+    else:
+        conv_layer = conv
+
+    kwargs = {'kernel_size': kernel_size,
+              'strides': strides,
+              'padding': padding,
+              'data_format': data_format,
+              'dilation_rate': dilation_rate,
+              'activation': activation,
+              **kwargs}
+
+    if transpose:
+        kwargs.pop('dilation_rate')
+
+    inputs_shape = inputs.get_shape().as_list()
+    axis = -1 if data_format == 'channels_last' else 1
+    size = [-1] * (dim + 2)
+    size[axis] = 1
+    channels_in = inputs_shape[axis]
+
+    depthwise_layers = []
+    for channel in range(channels_in):
+        start = [0] * (dim + 2)
+        start[axis] = channel
+
+        input_slice = tf.slice(inputs, start, size)
+
+        _kwargs = {**kwargs, 'inputs': input_slice, 'filters': depth_multiplier, 'name': 'slice-%d' % channel}
+
+        slice_conv = conv_layer(**_kwargs)
+        depthwise_layers.append(slice_conv)
+
+    # Concatenate the per-channel convolutions along the channel dimension.
+    depthwise_conv = tf.concat(depthwise_layers, axis=axis)
+
+    if channels_in * depth_multiplier != filters:
+        _kwargs = {**kwargs,
+                   'inputs': depthwise_conv,
+                   'filters': filters,
+                   'kernel_size': 1,
+                   'strides': 1,
+                   'name': 'pointwise'}
+        if not transpose:
+            kwargs['dilation_rate'] = 1
```

Comment:
_kwargs?
As 1 is a default values may be it is not needed at all.

Reply:
If dilation rate for previous conv layers was not 1 then we must rewrite it.